spec/spec_helper.rb

```diff
@@ -39,7 +39,9 @@
 begin
   # Ignore interpreter warnings from external libraries
   require 'warning'
-  Warning.ignore([:method_redefined, :not_reached, :unused_var], %r{.*/gems/[^/]*/lib/})
+  GEMS_PATH = %r{.*/gems/[^/]*/lib/}.freeze
+  Warning.ignore([:method_redefined, :not_reached, :unused_var, :arg_prefix], GEMS_PATH)
+  Warning.ignore(/circular require considered harmful/, GEMS_PATH)
```

Comment:
Not sure if you are familiar, but this uses the https://github.com/jeremyevans/ruby-warning#label-Examples gem to suppress errors that happen inside gems we depend on (`%r{.*/gems/[^/]*/lib/}` ensures we don't suppress warnings from our own files).

Reply:
- The closeness of the `GEMS_PATH` name with the `GEM_HOME` environment variable triggers some cognitive dissonance.
- Does `GEMS_PATH`  need to be a toplevel const, which would presumably make it global?
- The regex preexists this diff, but it feels like there would be room for false positive matches.
- It might also have false negatives if someone puts e.g a `path: ...` arg in the Gemfile.

I was not familiar with `ruby-warning`, but its README has the following as example, which would presumably achieve the same:

```
# Ignore all warnings in Gem dependencies
Gem.path.each do |path|
  Warning.ignore(//, path)
end
```

WDYT?

Reply:
I did what yo suggested.
The `Gem.path.each` is very clean 👍.

I did not exclude all gem warnings because some of them can be triggered by errors in our gem, for example when we we provide incorrect Hash object as the last argument in a method, instead of using a keyword splat `**opt`.